src/oni2-keyboard-layout/stubs/keyboard-layout.c

```diff
@@ -0,0 +1,25 @@
+#include <caml/alloc.h>
+#include <caml/callback.h>
+#include <caml/memory.h>
+#include <caml/mlvalues.h>
+#include <caml/threads.h>
+
+/* Create an OCaml Keymap Entry
+  Simple helper method to construct a keymap
+  entry from strings
+ */
+value createKeymapEntry(
+  const char *unmodified,
+  const char *withShift,
+  const char *withAltGraph,
+  const char *withAltGraphShift
+) {
+  CAMLparam0();
+  CAMLlocal1(keymapEntry);
+  keymapEntry = caml_alloc(4, 0);
+  Store_field(keymapEntry, 0, caml_copy_string(unmodified));
```

Comment:
Are there cases where these strings can be `NULL`? I'm wondering if we need to make these optional.

Reply:
The strings can be null, but the string would have length 0 -- we can make empty strings `None` and non-empty strings `Some(key)`!

Reply:
I ended up doing this in e531d4c:
https://github.com/onivim/oni2/blob/e531d4c8d4da96bed33a919569831673aa347492/src/oni2-keyboard-layout/stubs/keyboard-layout.c#L17-L26